analytics.js

```diff
@@ -983,6 +1000,14 @@ class Analytics {
     this.userTraits = {};
     this.groupId = "";
     this.groupTraits = {};
+    const { manuallyTrackSession } = this.session.sessionInfo;
+    this.session.sessionInfo = {};
+    if (this.session.autoTrack) {
+      this.session.startAutoTracking();
+    }
+    if (manuallyTrackSession) {
+      this.session.start();
+    }
```

Comment:
Move all of this into the session's reset API.

Reply:
Done.